File: Submissions/#0486_div3/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int n, k;
  cin >> n >> k;

  map<int, int> mp;

  for (int i = 0; i < n; i++) {
    int v;
    cin >> v;
    mp[v] = i;
  }

  if (k <= mp.size()) {
    
    cout << "YES" << endl;

    vector<int> answer(mp.size());
    int current = 0;
    
    for (auto item:mp) {
      answer[current] = item.second + 1;
      current++;
    }
    
    sort(answer.begin(), answer.end());
    
    for (int i = 0; i < k; i++) {
      cout << answer[i] << " ";
    }
    
    cout << endl;
  
  } else {
    cout << "NO" << endl;
  }

}

File: Submissions/#0486_div3/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int k = sc.nextInt();

    TreeMap<Integer, Integer> mp = new TreeMap<>();

    for (int i = 0; i < n; i++) {
      int v = sc.nextInt();
      mp.put(v, i);
    }
    sc.close();

    if (k <= mp.size()) {
    
      System.out.println("YES");
  
      int[] answer = new int[mp.size()];
      int current = 0;
      
      for (Integer value : mp.values()) {
        answer[current] = value + 1;
        current++;
      }
      
      Arrays.sort(answer);
      
      for (int i = 0; i < k; i++) {
        System.out.print(answer[i] + " ");
      }
      
      System.out.println("");
    
    } else {
      System.out.println("NO");
    }

  }

}